sdk/eventhub/event-hubs/src/eventHubProducerClient.ts

```diff
@@ -180,6 +223,10 @@ export class EventHubProducerClient {
    */
   async close(): Promise<void> {
     await this._client.close();
+
+    for (const pair of this._producersMap) {
+      await pair[1].close();
+    }
```

Comment:
The close() on the client will go and close all the open producers which is why I hadnt expliclity closed the producers in the _producersMap

Whats the reason for explicitly closing the producers from the map?

Reply:
I wanted to be explicit about it. However you've brought up a good point that we should do the client _last_, not first. I'll address that in a future PR since this is "harmless"